### PR TITLE
Add support for TLS setup for Ironic

### PR DESCRIPTION
**Pull request by maelk**
This PR introduces a TLS setup for Ironic. It requires the following PRs to be merged :

- https://github.com/metal3-io/cluster-api-provider-metal3/pull/133
- https://github.com/metal3-io/baremetal-operator/pull/631
- https://github.com/metal3-io/ironic-inspector-image/pull/62
- https://github.com/metal3-io/ironic-image/pull/198

### CODE REVIEW

File: 03_launch_mgmt_cluster.sh [hunk 1]
```diff
@@ -205,17 +220,27 @@ function launch_baremetal_operator() {
     if [ "${CAPM3_VERSION}" != "v1alpha3" ]; then
       kubectl create namespace metal3
     else
-      BMO_CONFIG="${BMOPATH}/deploy/default"
+      if [ "${IRONIC_TLS_SETUP}" == "true" ]; then
+        BMO_CONFIG="${BMOPATH}/deploy/default"
+      else
+        BMO_CONFIG="${BMOPATH}/deploy/tls"
```
Comment (kashifest) [inline 03_launch_mgmt_cluster.sh:226]: shouldn't it be the other way around?

File: 03_launch_mgmt_cluster.sh [hunk 2]
```diff
@@ -205,17 +220,27 @@ function launch_baremetal_operator() {
...
+      fi
       deploy_kustomization
     fi
 
     if [ "${EPHEMERAL_CLUSTER}" == "minikube" ]; then
-      BMO_CONFIG="${BMOPATH}/ironic-deployment/keepalived"
+      if [ "${IRONIC_TLS_SETUP}" == "true" ]; then
+        BMO_CONFIG="${BMOPATH}/ironic-deployment/keepalived"
+      else
+        BMO_CONFIG="${BMOPATH}/ironic-deployment/keepalived-tls"
```
Comment (kashifest) [inline 03_launch_mgmt_cluster.sh:236]: And this one also?

**Review by kashifest — CHANGES_REQUESTED** [review]

File: 03_launch_mgmt_cluster.sh (hunk 1 above)
Comment (maelk) [reply 03_launch_mgmt_cluster.sh:226]: good catch

File: 03_launch_mgmt_cluster.sh [hunk 3]
```diff
@@ -205,17 +220,27 @@ function launch_baremetal_operator() {
...
+        cp "${IRONIC_CACERT_FILE}" "${BMOPATH}/ironic-deployment/keepalived-tls/ca.crt"
+        cp "${IRONIC_CERT_FILE}" "${BMOPATH}/ironic-deployment/keepalived-tls/tls.crt"
+        cp "${IRONIC_KEY_FILE}" "${BMOPATH}/ironic-deployment/keepalived-tls/tls.key"
+      fi
       deploy_kustomization
     fi
 
-    update_images
-
```
Comment (fmuyassarov) [inline 03_launch_mgmt_cluster.sh:218]: This you have already done in #456 :)

Comment (maelk) [reply 03_launch_mgmt_cluster.sh]: rebased, thanks

File: 03_launch_mgmt_cluster.sh [hunk 4]
```diff
@@ -211,12 +280,31 @@ function launch_baremetal_operator() {
...
     fi
 
     if [ "${EPHEMERAL_CLUSTER}" == "minikube" ]; then
-      BMO_CONFIG="${BMOPATH}/ironic-deployment/keepalived"
+      if [ "${IRONIC_TLS_SETUP}" != "true" ]; then
+        BMO_CONFIG="${BMOPATH}/ironic-deployment/keepalived"
+      else
+        BMO_CONFIG="${BMOPATH}/ironic-deployment/tls/keepalived"
+        cp "${IRONIC_CACERT_FILE}" "${BMOPATH}/ironic-deployment/tls/keepalived/ironic-ca.crt"
+        cp "${IRONIC_CACERT_FILE}" "${BMOPATH}/ironic-deployment/tls/keepalived/ironic-inspector-ca.crt"
```
Comment (fmuyassarov) [inline 03_launch_mgmt_cluster.sh:300]:
```suggestion
        cp "${IRONIC_INSPECTOR_CACERT_FILE}" "${BMOPATH}/ironic-deployment/tls/keepalived/ironic-inspector-ca.crt"
```
?

Comment (maelk) [reply 03_launch_mgmt_cluster.sh]: done

**Pull request closed without merging**